flang/lib/Lower/ReductionRuntime.cpp

```diff
@@ -86,6 +86,121 @@ struct ForcedMinvalInteger16 {
   }
 };
 
+/// Placeholder for real*10 version of Product Intrinsic
+struct ForcedProductReal10 {
+  static constexpr const char *name = ExpandAndQuoteKey(RTNAME(ProductReal10));
+  static constexpr Fortran::lower::FuncTypeBuilderFunc getTypeModel() {
+    return [](mlir::MLIRContext *ctx) {
+      auto ty = mlir::FloatType::getF80(ctx);
+      return mlir::FunctionType::get(ctx, {ty}, {ty});
```

Comment:
Have you tested generating those ? The function type here does not look correct to me. `ProductReal10` takes a descriptor and a couple more arguments, so the last argument of `mlir::FunctionType::get` should not be `{ty}`.
Same comment apply for all added `Forcedxxx` functions.

Reply:
Yes, the arguments were not correct. It's now correct and there are complex(10) versions of the regression tests.